Lib/functools.py

```diff
@@ -16,7 +16,15 @@
 try:
     from _functools import reduce
 except ImportError:
-    pass
+    def reduce(function, iterable, initializer=None):
+        it = iter(iterable)
+        if initializer is None:
+            value = next(it)
```

Comment:
This raises `StopIteration` when it should raise `TypeError: reduce() of empty sequence with no initial value`